### PR TITLE
Resolve Show exam questions

### PR DESCRIPTION
**Pull request by HaiTranVar**

### CODE REVIEW

File: app/views/tests/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<h2 class="text-left"><%= @name %></h2>
+<h4>Times: <%= @time %>s</h4>
+<h4><%= link_to "Back", root_path %></h4>
+<ul>
```
Comment (mai-123) [inline app/views/tests/show.html.erb]: ol instead ul

File: config/routes.rb [hunk 2]
```diff
@@ -3,6 +3,8 @@
   get "/login", to: "sessions#new"
   post "/login", to: "sessions#create"
   get "/logout", to: "sessions#destroy"
+  get "/show", to: "tests#show"
```
Comment (mai-123) [inline config/routes.rb]: remove this line

File: config/routes.rb [hunk 3]
```diff
@@ -3,6 +3,8 @@
...
   resources :password_resets, only: [:new, :create, :edit, :update]
   resources :users
+  resources :tests
```
Comment (mai-123) [inline config/routes.rb]: add only with reasonable options (show, index)

File: db/migrate/20190716042020_create_questions.rb [hunk 4]
```diff
@@ -2,11 +2,11 @@ class CreateQuestions < ActiveRecord::Migration[5.2]
...
     create_table :questions do |t|
       t.string :content
-      t.integer :test_id
+      t.integer :test_id, foreign_key: true
 
       t.timestamps
     end
 
-    add_index :questions, :test_id
+    add_index :questions, [:test_id, :created_at]
```
Comment (mai-123) [inline db/migrate/20190716042020_create_questions.rb]: remove

File: app/assets/stylesheets/custom.scss [hunk 5]
```diff
@@ -61,3 +61,12 @@ header ul a {
 .header.text-right {
     margin: 10px auto;
 }
+
+.question {
+    margin: 10px 0;
```
Comment (ducminhv) [inline app/assets/stylesheets/custom.scss]: indent

File: app/assets/stylesheets/custom.scss [hunk 6]
```diff
@@ -61,3 +61,12 @@ header ul a {
...
+}
+
+span {
+    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
```
Comment (ducminhv) [inline app/assets/stylesheets/custom.scss]: indent

File: app/controllers/questions_controller.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class QuestionsController < ApplicationController
+  before_action :get_question
+  def show
+    @questions = @test.questions
+  end
+
+  private
+  def get_question
+    @test = Test.find(1)
```
Comment (ducminhv) [inline app/controllers/questions_controller.rb]: ....

File: app/controllers/questions_controller.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+class QuestionsController < ApplicationController
+  before_action :get_question
+  def show
```
Comment (ducminhv) [inline app/controllers/questions_controller.rb]: empty line before this line

File: app/controllers/questions_controller.rb [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+    @questions = @test.questions
```
Comment (ducminhv) [inline app/controllers/questions_controller.rb]: includes :answers

File: app/controllers/tests_controller.rb [hunk 10]
```diff
@@ -23,7 +24,21 @@ def create
     end
   end
 
+  def show
+    @questions = @test.questions
+    @name = @test.name
```
Comment (ducminhv) [inline app/controllers/tests_controller.rb]: dont need this variable

File: app/controllers/tests_controller.rb [hunk 11]
```diff
@@ -23,7 +24,21 @@ def create
...
+    @time = @test.time
```
Comment (ducminhv) [inline app/controllers/tests_controller.rb]: dont need this variable

File: app/models/question.rb [hunk 12]
```diff
@@ -1,5 +1,7 @@
 class Question < ApplicationRecord
   belongs_to :test
 
-  validates :test, presence: true
+  has_many :answers
+
+  validates :test_id, presence: true
```
Comment (ducminhv) [inline app/models/question.rb]: `validates :test, presence: true`

File: app/views/tests/show.html.erb [hunk 13]
```diff
@@ -0,0 +1,19 @@
+<h2 class="text-left"><%= @name %></h2>
+<h4>Times: <%= @time %>s</h4>
```
Comment (ducminhv) [inline app/views/tests/show.html.erb]: i18n

File: app/views/tests/show.html.erb [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+<h4><%= link_to "Back", root_path %></h4>
```
Comment (ducminhv) [inline app/views/tests/show.html.erb]: i18n

File: app/views/tests/show.html.erb [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+<ol>
+  <% @questions.each do |question| %>
+  <li class="question", type="1">
+    <span><%= question.content %></span>
+    <ol type="A">
+      <% question.answers.each do |answer| %>
+        <% if answer.is_correct? %>
+          <li><font color = "#337ab7"><%= answer.content %></font></li>
```
Comment (ducminhv) [inline app/views/tests/show.html.erb]: set in css file

File: db/migrate/20190716042020_create_questions.rb [hunk 16]
```diff
@@ -2,11 +2,11 @@ class CreateQuestions < ActiveRecord::Migration[5.2]
   def change
     create_table :questions do |t|
       t.string :content
-      t.integer :test_id
+      t.integer :test_id, foreign_key: true
```
Comment (ducminhv) [inline db/migrate/20190716042020_create_questions.rb]: remove this change

File: db/migrate/20190716042020_create_questions.rb [hunk 17]
```diff
@@ -2,11 +2,11 @@ class CreateQuestions < ActiveRecord::Migration[5.2]
...
 
       t.timestamps
     end
 
-    add_index :questions, :test_id
+    add_index :questions, [:test_id]
```
Comment (ducminhv) [inline db/migrate/20190716042020_create_questions.rb]: remove this change

File: app/models/test.rb [hunk 18]
```diff
@@ -4,7 +4,7 @@ class Test < ApplicationRecord
   enum kind: CATEGORY_TYPE
 
   has_many :results
-  has_many :questions
+  has_many :questions, dependent: :destroy
```
Comment (ducminhv) [inline app/models/test.rb]: remove this

File: app/assets/stylesheets/custom.scss [hunk 19]
```diff
@@ -59,5 +59,16 @@ header ul a {
...
 
 .header.text-right {
-    margin: 10px auto;
+  margin: 10px auto;
+}
+
+.question {
+  margin: 10px 0;
+}
+.corect {
```
Comment (ducminhv) [inline app/assets/stylesheets/custom.scss]: remove this

**Pull request merged**